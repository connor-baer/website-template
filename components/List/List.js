import { css } from '@emotion/core';
import styled from '@emotion/styled';

const UNORDERED = 'unordered';
const ORDERED = 'ordered';

const baseStyles = ({ theme }) => css`
  margin: 0;
  padding-left: 0;
  list-style-type: none;

  li {
    position: relative;
    padding-left: 2rem;

    &::before {
      position: absolute;
      left: 0;
      top: 0.25em;
      text-align: right;
      color: ${theme.colors.p500};
      display: inline-block;
      min-width: 1.25em;
      margin-left: 0rem;
      margin-right: 0.5rem;
    }
  }

  p {
    margin: 0;
  }
`;

const unorderedStyles = ({ type = UNORDERED }) =>
  type === UNORDERED &&
  css`
    li::before {
      content: 'â€¢';
      transform: scale(1.25);
      transform-origin: right;
    }
  `;

const orderedStyles = ({ type = UNORDERED }) =>
  type === ORDERED &&
  css`
    counter-reset: li;

    li {
      counter-increment: li;

      &::before {
        content: counter(li, decimal-leading-zero);
        font-variant-numeric: tabular-nums;
      }
    }
  `;

const List = styled('ul')(baseStyles, unorderedStyles, orderedStyles);

List.UNORDERED = UNORDERED;
List.ORDERED = ORDERED;

export default List;
