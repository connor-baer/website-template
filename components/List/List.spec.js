import React from 'react';

import List from './List';

describe('List', () => {
  function renderList(fn, props = {}) {
    return fn(<List {...props} />);
  }

  const defaultProps = {
    children: (
      <>
        <li>One</li>
        <li>Two</li>
      </>
    )
  };

  it('should render with default styles', () => {
    const { container } = renderList(render, { ...defaultProps });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with ordered styles', () => {
    const { container } = renderList(render, {
      ...defaultProps,
      type: List.ORDERED
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const html = renderList(renderToHtml, { ...defaultProps });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
