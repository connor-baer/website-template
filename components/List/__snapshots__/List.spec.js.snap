// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render with default styles 1`] = `
.circuit-0 {
  margin: 0;
  padding-left: 0;
  list-style-type: none;
}

.circuit-0 li {
  position: relative;
  padding-left: 2rem;
}

.circuit-0 li::before {
  position: absolute;
  left: 0;
  top: 0.25em;
  text-align: right;
  color: #f06595;
  display: inline-block;
  min-width: 1.25em;
  margin-left: 0rem;
  margin-right: 0.5rem;
}

.circuit-0 p {
  margin: 0;
}

.circuit-0 li::before {
  content: 'â€¢';
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
  -webkit-transform-origin: right;
  -ms-transform-origin: right;
  transform-origin: right;
}

<ul
  class="circuit-0 circuit-1"
>
  <li>
    One
  </li>
  <li>
    Two
  </li>
</ul>
`;

exports[`List should render with ordered styles 1`] = `
.circuit-0 {
  margin: 0;
  padding-left: 0;
  list-style-type: none;
  counter-reset: li;
}

.circuit-0 li {
  position: relative;
  padding-left: 2rem;
}

.circuit-0 li::before {
  position: absolute;
  left: 0;
  top: 0.25em;
  text-align: right;
  color: #f06595;
  display: inline-block;
  min-width: 1.25em;
  margin-left: 0rem;
  margin-right: 0.5rem;
}

.circuit-0 p {
  margin: 0;
}

.circuit-0 li {
  counter-increment: li;
}

.circuit-0 li::before {
  content: counter(li,decimal-leading-zero);
  font-variant-numeric: tabular-nums;
}

<ul
  class="circuit-0 circuit-1"
  type="ordered"
>
  <li>
    One
  </li>
  <li>
    Two
  </li>
</ul>
`;
